scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -614,6 +645,70 @@ cdef _dijkstra_directed(
 
     free(nodes)
 
+cdef _dijkstra_directed_multi(
+            np.ndarray[ITYPE_t, ndim=1, mode='c'] source_indices,
+            np.ndarray[DTYPE_t, ndim=1, mode='c'] csr_weights,
+            np.ndarray[ITYPE_t, ndim=1, mode='c'] csr_indices,
+            np.ndarray[ITYPE_t, ndim=1, mode='c'] csr_indptr,
+            np.ndarray[DTYPE_t, ndim=1, mode='c'] dist_matrix,
+            np.ndarray[ITYPE_t, ndim=1, mode='c'] pred,
```

Comment:
I suspect there's probably a preference for typed memoryviews in new Cython code, although you'd be correct in noticing that there are cases of the older buffer usage in this module

Reply:
I'll work on changing this code style tomorrow and post a note when I think its done.

Reply:
I believe I have integrated the memory views, although line 344 errored out when i changed the signature of _floyd_warshall, so I have left that function in the old style.  I'm not experienced enough with memoryarray's to be sure why.